datasets/multi_eurlex/multi_eurlex.py

```diff
@@ -0,0 +1,183 @@
+               classification dataset for zero-shot cross-lingual transfer},
+  booktitle = {Proceedings of the 2021 Conference on Empirical Methods
+               in Natural Language Processing},
+  year = {2021},
+  publisher = {Association for Computational Linguistics},
+  location = {Punta Cana, Dominican Republic},
+}"""
+
+_DESCRIPTION = """\
+MultiEURLEX comprises 65k EU laws in 23 official EU languages (some low-ish resource).
+Each EU law has been annotated with EUROVOC concepts (labels) by the Publication Office of EU.
+As with the English EURLEX, the goal is to predict the relevant EUROVOC concepts (labels);
+this is multi-label classification task (given the text, predict multiple labels).
+"""
+
+DATA_URL = "https://zenodo.org/record/5363165/files/multi_eurlex.tar.gz"
+
+_LANGUAGES = [
+    "en",
+    "da",
+    "de",
+    "nl",
+    "sv",
+    "bg",
+    "cs",
+    "hr",
+    "pl",
+    "sk",
+    "sl",
+    "es",
+    "fr",
+    "it",
+    "pt",
+    "ro",
+    "et",
+    "fi",
+    "hu",
+    "lt",
+    "lv",
+    "el",
+    "mt",
+]
+
+
+class MultiEURLEXConfig(datasets.BuilderConfig):
+    """BuilderConfig for MultiEURLEX."""
+
+    def __init__(self, language: str, languages=None, label_level="level_1", **kwargs):
+        """BuilderConfig for MultiEURLEX.
+
+        Args:
+        language: One of ar,bg,de,el,en,es,fr,hi,ru,sw,th,tr,ur,vi,zh, or all_languages
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(MultiEURLEXConfig, self).__init__(**kwargs)
+        self.language = language
+        self.label_level = label_level
+        if language != "all_languages":
+            self.languages = [language]
+        else:
+            self.languages = languages if languages is not None else _LANGUAGES
+
+
+class MultiEURLEX(datasets.GeneratorBasedBuilder):
+    """MultiEURLEX - A multi-lingual and multi-label legal document classification dataset. Version 1.0"""
+
+    VERSION = datasets.Version("1.0.0", "")
+    BUILDER_CONFIG_CLASS = MultiEURLEXConfig
+    BUILDER_CONFIGS = [
+        MultiEURLEXConfig(
+            name=lang,
+            language=lang,
+            version=datasets.Version("1.0.0", ""),
+            description=f"Plain text import of MultiEURLEX for the {lang} language",
+        )
+        for lang in _LANGUAGES
+    ] + [
+        MultiEURLEXConfig(
+            name="all_languages",
+            language="all_languages",
+            version=datasets.Version("1.0.0", ""),
+            description="Plain text import of MultiEURLEX for all languages",
+        )
+    ]
+
+    def _info(self):
+        if self.config.language == "all_languages":
+            features = datasets.Features(
+                {
+                    "celex_id": datasets.Value("string"),
+                    "text": datasets.Translation(
+                        languages=_LANGUAGES,
+                    ),
+                    "labels": datasets.features.Sequence(datasets.Value("string")),
```

Comment:
Here you can use the `ClassLabel` feature type, with the names of all the labels

Reply:
I would prefer to keep the labels as a list of the original label ids (str), if this is not a big issue.